src/coreclr/jit/lsraarm64.cpp

```diff
@@ -1481,6 +1481,21 @@ int LinearScan::BuildHWIntrinsic(GenTreeHWIntrinsic* intrinsicTree, int* pDstCou
         {
             srcCount += BuildContainedCselUses(containedCselOp, delayFreeOp, candidates);
         }
+        else if ((intrin.category == HW_Category_SIMDByIndexedElement) && (genTypeSize(intrin.baseType) == 2) && !HWIntrinsicInfo::HasImmediateOperand(intrin.id))
```

Comment:
Thanks for finding this. To keep in the existing style, could you:

Expand `getDelayFreeOperand()` to:
`GenTree* LinearScan::getDelayFreeOperand(GenTreeHWIntrinsic* intrinsicTree, bool embedded, bool *forceDelay)`
Where `forceDelay` is a return argument.

Move your `else if` into `getDelayFreeOperand()` (I guess it'll go into the default case) and if it passes, set `*forceDelay=true` and return nullptr.

Then in line 1500 ("Only build as delay free use if register types match") add a `|| forceDelay == true`

Reply:
Yes, I hadn't really taken the `for` loop into account properly.

For all these `SIMDByIndexedElement` intrisnics, will `op1` always be a delay operand?

Reply:
If you see the code, it's not delayed as such, but rather is a way for us to tell that it is the preferred register for target. It is not marked as delay-free.